pkg/scheduler/schedule_one.go

```diff
@@ -1128,6 +1130,9 @@ func updatePod(ctx context.Context, client clientset.Interface, pod *v1.Pod, con
 	}
 	if nnnNeedsUpdate {
 		podStatusCopy.NominatedNodeName = nominatingInfo.NominatedNodeName
+		if utilfeature.DefaultFeatureGate.Enabled(features.PodObservedGenerationTracking) {
```

Comment:
same comment about testing what happens if the scheduler tries to set the field and the API server clears it, to avoid a hotloop / infinite repatch retry

Reply:
https://github.com/kubernetes/kubernetes/pull/130352#discussion_r1977730000